index.html

```diff
@@ -787,36 +787,154 @@ <h4>`ThingLink`</h4>
+                and security bootstrapping would not be required.  
+                </p>
+                <p>Security bootstrapping is also only (potentially) necessary on the <em>first</em> access to
+                a TD. Once a Discoverer has determined what credentials and authentication mechanism
+                are required to access a particular exploration service, they can retain them
+                and use them for future accesses.
+                </p>
+                <p>
+                <span class="rfc2119-assertion" id="security-bootstrapping-endpoints">Security
+                bootstrapping MUST be provided on any HTTP endpoint that serves a TD.</span>
+                This applies not only to endpoints serving TDs from self-describing Things,
+                but also to endpoints such as <code>things/{id}</code> in <a>WoT Thing Description Directory</a>
+                exploration services.  
+                </p>
+                <p>In the HTTP protocol, authentication can generally be negotiated by
+                the HTTP server returning a "401 (Unauthorized)" response code in conjunction with
+                a `WWW-Authenticate` header that specifies the information required for
+                authentication.  
+                To gain access, the client then needs to make another request 
+                with the necessary information.
+                <a href="https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml">There 
+                are several authentication schemes registered with IANA.</a>
+                Not all of these are in wide use, some are experimental, and there is only
+                partial overlap with the schemes supported by TDs.
+                In addition, note that the <code>oauth</code> scheme in the IANA registration refers
+                to OAuth1, which is deprecated, so it should not be used.
+                The relevant OAuth2 flow, the <code>code</code> flow, instead 
+                begins with a "302 (Redirection)" to an authentication
+                server, eventually resulting in credentials (bearer tokens in the case of WoT)
+                that can be used for access.
+                To enable security bootstrapping on a wide variety of devices as well
+                as on browsers, the following constraints should be observed:
+                </p>
+                <ul>
+                <li><span class="rfc2119-assertion" id="exploration-secboot-401">
+                After initial contact with an exploration service using the URL provided
+                by an introduction mechanism, the exploration service
+                MUST reply with either an HTTP "401 (Unauthorized)" response code 
+                or and HTTP "302 (Redirection)" response code (OAuth2 only) if appropriate
+                authentication information has not been provided but access can be granted
+                when it is.</span>
```

Comment:
Again, having a MUST here limits us to a few mechanism that we know of. 

1. The oauth2 spec allows other redirection mechanisms: https://datatracker.ietf.org/doc/html/rfc6749#section-1.7
2. I've mentioned mTLS before (where the client authenticates using a certificate in TLS negotiation) which is very secure, supported by browsers and HTTP clients, and very relevant for human to machine (e.g. with a smart card) and machine to machine (certificate issued by vendor and installed on IoT devices) interaction. I'm not sure that 401/302 covers it. 
https://www.cloudflare.com/learning/access-management/what-is-mutual-tls/
https://medium.com/@sevcsik/authentication-using-https-client-certificates-3c9d270e8326

Reply:
I think the issue with mTLS is privacy, so it's not a silver bullet.  I'm not against including but would like to discuss it separately, as it may also impact the TD spec.